### ISSUE TITLE
Error on received attestation

### ISSUE THREAD

**Issue by ajsutton** [Consensys-Incorporated/teku#2143]
### Description
Got a series of errors like this one on Onyx:

```
"class":"AbstractRouter","message":"Exception while handling message from peer io.libp2p.etc.util.P2PServiceSemiDuplex$SDPeerHandler@16f6ba84: from: "\000%\b\002\022!\002S\267o\023\214B/\237[E\302\2059\3548\254\001\375i\345\270\177\223\304\177\036-B\215S)R"
data: "\365\001\030\344\000\000\000\200\001\0002\001\000\200\202\246\3671\274\266\332\032u\346\314\365\376\262\301 <\233\004\024x\301Z\265\336{\005\177\363I~\234\v\r.\200/\2008n\260jS\372m\343\233\r\250\311\2410\251\354\227ka\002\275?Z\365\302s\252\005\272\311\f\r(~P\000\360p\220\020\220\317G-\344&6\f\025\305e\224qm\031\233\307\024sd\302\027kV9TO\347hG->\261\342\302\347\240\021I\221\275\033^\364\271{\n\035T#\324\330A%T\336\375\366\266\"\036\224\3241\244\3401HA\376q\305w\323\265a!\356X\r\353\357\374\004\366\201s\365\363n\315+\223\215\000\200\000\000\000\000\000\000\000\000\000\000\000\000\000\000\001"
seqno: "\026\030L\034E?\226\374"
topicIDs: "/eth2/a65b4897/beacon_attestation_0/ssz_snappy"
","throwable":" java.util.concurrent.ExecutionException: java.lang.UnsupportedOperationException: Empty collection can't be reduced.
at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
at io.libp2p.pubsub.AbstractRouter.onInbound(AbstractRouter.kt:209)
at io.libp2p.etc.util.P2PService.streamInbound(P2PService.kt:149)
at io.libp2p.etc.util.P2PService$StreamHandler$channelRead$1.invoke(P2PService.kt:49)
at io.libp2p.etc.util.P2PService$StreamHandler$channelRead$1.invoke(P2PService.kt:35)
at io.libp2p.etc.util.P2PService$sam$java_lang_Runnable$0.run(P2PService.kt)
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)\nCaused by: java.lang.UnsupportedOperationException: Empty collection can't be reduced.
at io.libp2p.pubsub.PubsubApiImpl$onNewMessage$1.invoke(PubsubApiImpl.kt:130)
at io.libp2p.pubsub.PubsubApiImpl$onNewMessage$1.invoke(PubsubApiImpl.kt:23)
at io.libp2p.etc.types.AsyncExtKt$thenApplyAll$1.apply(AsyncExt.kt:84)
at io.libp2p.etc.types.AsyncExtKt$thenApplyAll$1.apply(AsyncExt.kt)
at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:680)
at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:658)
at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2094)
at io.libp2p.etc.types.AsyncExtKt.thenApplyAll(AsyncExt.kt:79)
at io.libp2p.pubsub.PubsubApiImpl.onNewMessage(PubsubApiImpl.kt:73)
at io.libp2p.pubsub.PubsubApiImpl.access$onNewMessage(PubsubApiImpl.kt:23)
at io.libp2p.pubsub.PubsubApiImpl$1.invoke(PubsubApiImpl.kt:55)
at io.libp2p.pubsub.PubsubApiImpl$1.invoke(PubsubApiImpl.kt:23)
at io.libp2p.pubsub.AbstractRouter.onInbound(AbstractRouter.kt:194)\n\t... 10 more\n"}
```

**Comment by ajsutton** [comment]
Now also seeing this on Witti.

**Comment by benjaminion** [comment]
Also started seeing these on Witti today.

They began appearing almost immediately after my node stopped keeping up with the chain head: progress stalled from this point until I shut down 3 hours later. It may or may not be related, but is suspicious.

**Comment by benjaminion** [comment]
My Witti log is flooded with these today (literally thousands of them), although everything seems to be running smoothly.

```
$ ./build/install/teku/bin/teku --version
teku/v0.11.5-dev-dbe57ded/linux-x86_64/-privatebuild-openjdk64bitservervm-java-13
```

**Comment by benjaminion** [comment]
Got another 18578 of these already in today's Witti log file, and it's not even midday yet. They are basically making the logs unusable.

**Comment by Nashatyrev** [comment]
`java.lang.UnsupportedOperationException: Empty collection can't be reduced.` error happened somewhere in gossip attestation validation code. 
I'd expect to see `ExecutionException` cause here as well. Strange it is not printed

**Comment by Nashatyrev** [comment]
Created function to parse back the protobuf binary dump

```kt
    fun parse(str: String): ByteArray {
        val bytes = mutableListOf<Byte>()
        var pos = 0
        while (pos < str.length) {
            bytes += when (str[pos]) {
                '\\' -> {
                    pos++
                    when (str[pos]) {
                        in '0'..'9' -> {
                            val r = ((("" + str[pos]).toInt() shl 6) or
                                    (("" + str[pos + 1]).toInt() shl 3) or
                                    (("" + str[pos + 2]).toInt())).toByte()
                            pos += 2
                            r
                        }
                        'a' -> 0x07
                        'b' -> '\b'.toByte()
                        'f' -> 0xC
                        'n' -> '\n'.toByte()
                        'r' -> '\r'.toByte()
                        't' -> '\t'.toByte()
                        'v' -> 0x0b
                        '\\' -> '\\'.toByte()
                        '\'' -> '\''.toByte()
                        '"' -> '"'.toByte()
                        else -> throw IllegalArgumentException("Invalid escape char")
                    }.also { pos++ }
                }
                else -> str[pos++].toByte()
            }
        }
        return bytes.toByteArray()
    }
```

got this hex `data` field content: 

```
f50118e40000008001003201008082a6f731bcb6da1a75e6ccf5feb2c1203c9b041478c15ab5de7b057ff3497e9c0b0d2e802f80386eb06a53fa6de39b0da8c9a130a9ec976b6102bd3f5af5c273aa05bac90c0d287e5000f070901090cf472de426360c15c56594716d199bc7147364c2176b5639544fe768472d3eb1e2c2e7a0114991bd1b5ef4b97b0a1d5423d4d8412554defdf6b6221e94d431a4e0314841fe71c577d3b56121ee580debeffc04f68173f5f36ecd2b938d0080000000000000000000000000000001
```

**Comment by ajsutton** [comment]
The line numbers in the exception really don't line up very well.  But in `PubsubApiImpl.onNewMessage` which appears in the stack trace, it has a call to `reduce` without checking if there are no entries:
```
val validationFuts = synchronized(this) {
            msg.topicIDsList.mapNotNull { subscriptions[Topic(it)] }.flatten().distinct()
        }.map {
            it.receiver.apply(rpc2Msg(msg))
        }
return validationFuts.thenApplyAll { it.reduce(validationResultReduce) }
```

and that would result in the `UnsupportedOperationException` we're seeing.  What I can' tell is why there's nothing in the list being reduced.  From what I can understand of the Kotlin std lib we'd get that exception if `validationFuts` winds up empty, which would happen if there are no subscriptions for the topic. Could this be a race condition where we're being sent messages for a topic that we've unsubscribed from after the subscriber has been removed from the list but before other peers have recognised that we've unsubscribed?  Or possibly just that peers are sending us messages that we aren't interested in...
